Source/NSubstitute/Core/CallSpecificationFactory.cs

```diff
@@ -15,7 +15,7 @@ public ICallSpecification CreateFrom(ICall call, MatchArgs matchArgs)
         {
             var methodInfo = call.GetMethodInfo();
             var argumentSpecs = call.GetArgumentSpecifications();
-            var arguments = call.GetArguments();
+            var arguments = call.GetOriginalArguments();
```

Comment:
@Zvirja This has always been a problem with out/ref in cases where we set the values from within a `Returns`. Does your changes fix that problem overall, or just for the custom handlers? (I can check it myself, but thought you might have the info at hand :) )